### PR TITLE
Improve merge performance

### PR DESCRIPTION
**Pull request by hrsh7th**
As title.

### CODE REVIEW

File: autoload/jetpack.vim [hunk 1]
```diff
@@ -73,7 +73,7 @@ endfunction
 
 function s:mergable(pkgs, pkg)
   let path = []
-  for p in path
+  for p in a:pkgs
```
Comment (hrsh7th) [inline autoload/jetpack.vim:76]: It was just wrong, I think.

File: autoload/jetpack.vim [hunk 2]
```diff
@@ -202,26 +202,19 @@ function jetpack#update(...)
...
-    if g:jetpack#optimization >= 1
-         \ && !pkg.opt
-         \ && (g:jetpack#optimization == 2 || s:mergable(bundle, pkg))
+    if g:jetpack#optimization >= 1 && !pkg.opt
       call add(bundle, pkg)
     else
       call add(unbundle, pkg)
     endif
-    call s:setbufline(i+3, printf('Checked %s', pkg.name))
   endfor
```
Comment (hrsh7th) [inline autoload/jetpack.vim:214]: First, we should classify bundles / unbundles simply.

File: autoload/jetpack.vim [hunk 3]
```diff
@@ -230,14 +223,30 @@ function jetpack#bundle()
...
+        if filereadable(destfile)
+          let ignore = v:true
+          break
+        endif
+      endfor
+      if ignore
+        call add(unbundle, pkg)
+        continue
       endif
+    endif
```
Comment (hrsh7th) [inline autoload/jetpack.vim:241]: Check the `destfile` exists or not at the run-time.

**Comment by tani** [comment]
Thank you for your contribution. It looks fine!

**Pull request merged**